src/App.tsx

```diff
@@ -1,27 +1,86 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+import { Loader } from './components/Loader';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleClick = useCallback((buttonName: string) => {
+    setIsLoading(true);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    let promise: Promise<Good[]>;
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    switch (buttonName) {
+      case 'all-button':
+        promise = getAll();
+        break;
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      case 'first-five-button':
+        promise = get5First();
+        break;
+
+      case 'red-button':
+        promise = getRedGoods();
+        break;
+
+      default:
+        promise = Promise.reject(new Error('Unexpected case in switch'));
+        break;
```

Comment:
Instead of `switch...case` better pass a reference to the loader function as an argument. It's way more flexible and easy to maintain.